src/test/kotlin/com/gamingbacklog/api/gamingbacklogapi/clients/IGDBClientTests.kt

```diff
@@ -1,4 +1,169 @@
 package com.gamingbacklog.api.gamingbacklogapi.clients
 
+import com.gamingbacklog.api.gamingbacklogapi.testutil.models.MockResponse
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.mockito.Mockito.mock
+import org.mockito.kotlin.given
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.context.annotation.Import
+import org.springframework.test.context.TestPropertySource
+
+import org.mockito.kotlin.any
+import org.mockito.kotlin.eq
+
+@SpringBootTest
+@Import(IGDBClient::class)
+@TestPropertySource(properties = ["CLIENT_ID=test_id", "CLIENT_SECRET=test_secret"])
 class IGDBClientTests {
+  lateinit var externalAPICall: ExternalAPICall
+  lateinit var igdbClient: IGDBClient
+
+  @BeforeEach
+  fun configureSystem() {
+    externalAPICall = mock(ExternalAPICall::class.java)
+    igdbClient = IGDBClient(externalAPICall)
+  }
+
+  @Nested
+  @DisplayName("Tests for Authenticate")
+  inner class Authenticate {
+
+    @Test
+    fun shouldSuccessfullyReturnToken() {
+      given(externalAPICall.postExternalCall(any(), eq(null), eq(null))).willReturn(
+        MockResponse(
+          "{\"access_token\":\"test_secret\",\"expires_in\":5180593,\"token_type\":\"bearer\"}",
+          200
+      ))
+      val result = igdbClient.authenticate()
+      Assertions.assertEquals("test_secret", result.access_token)
+      Assertions.assertEquals("5180593", result.expires_in)
+      Assertions.assertEquals("bearer", result.token_type)
+    }
+
+    @Test
+    fun shouldErrorOnAuthentication() {
+      given(externalAPICall.postExternalCall(any(), eq(null), eq(null))).willReturn(
+        MockResponse(
+          "{\"status\":400,\"message\":\"invalid\"}",
+          400
+        )
+      )
+      val result = igdbClient.authenticate()
+      Assertions.assertEquals("", result.access_token)
+      Assertions.assertEquals("", result.expires_in)
+      Assertions.assertEquals("", result.token_type)
+    }
+  }
+
+  @Nested
+  @DisplayName("Tests for Games Request")
+  inner class GamesRequest {
+    // successful return of game
+    @Test
+    fun shouldSuccessfullyReturnGame() {
+      given(externalAPICall.postExternalCall(any(), any(), any())).willReturn(
+        MockResponse(
```

Comment:
Same with this - it would be great if we could define a Kotlin class for these API responses to avoid these verbose strings in the future. We'd have to add in (de)serialization logic as well for that

Alternatively as a simpler short term approach, we could define a constants file for common use test strings to clean this up